src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -827,18 +827,25 @@ private void endAppend() {
   private void commitFile() {
     log.debug("Committing files");
     appended.clear();
+
+    long latestCommitted = -1;
     for (String encodedPartition : tempFiles.keySet()) {
-      commitFile(encodedPartition);
+      long endOffset = commitFile(encodedPartition);
+      latestCommitted = Math.max(latestCommitted, endOffset);
+    }
```

Comment:
How about?
`long lastCommitted = tempFiles.keySet().stream().mapToLong(this::commitFile).max().orElse(-1);`